lib/ProductOpener/Export.pm

```diff
@@ -624,5 +594,43 @@ sub export_csv($) {
 }
 
 
+sub include_image_paths($$$) {
+	my $product_ref = shift;
+	my %populated_fields = shift;
+	my %other_images = shift;
```

Comment:
Same issue as in [#7157](https://github.com/openfoodfacts/openfoodfacts-server/pull/7157#discussion_r935369936), hashes can't be passed like that.

Reply:
Would it work if we used the perl signature instead?
Like:
```
sub include_image_paths($product_ref, %populated_fields, %other_images) {
```
Or is there no way around it?

Reply:
With signatures, there can be a single hash as a final argument that slurps remaining arguments as key-value pairs, but not two.

References are the way to go with passing a hash to a subroutine. See [perlreftut](https://perldoc.perl.org/perlreftut) and [perlref](https://perldoc.perl.org/perlref) for how to work with refs.

There's also no need to return the values when they're references.

Reply:
@Ban3 is right, you need to pass references to the hashes, instead of the hashes themselves.
The first argument works like this, we pass $product_ref (the _ref at the end is a naming convention to indicate that it is a reference) which a reference to a product hash. So instead of passing %some_hash, we pass $some_hash_ref. You can get a reference for a hash like this:  $my_hash_ref = \%my_hash.
Then when you read or write $my_hash_ref->{key}, it's like reading or writing %my_hash{key}.

Reply:
> You can get a reference for a hash like this: $my_hash_ref = %my_hash.

The backslash probably got interpreted as escape character by Markdown, should be `$my_hash_ref = \%my_hash`.

Reply:
Would it be like this?
```
sub include_image_paths($product_ref, $populated_fields_ref, $other_images_ref) {
        my $populated_fields_ref = %populated_fields;
        my $other_images_ref = %other_images;
```
Or would it be outside of the function?

Reply:
Ok, thank you